scripts/pre_commit_linter.py

```diff
@@ -342,10 +356,48 @@ def _pre_commit_linter():
     summary_messages.append(py_result.get())
     print '\n'.join(summary_messages)
     print ''
+    return all_files, summary_messages
 
+
+# This part is used for detecting bad patterns.
+def _check_bad_patterns(all_files, summary_messages):
+    print 'Starting Pattern Checks'
+    print '----------------------------------------'
+    total_files_checked = 0
+    total_error_count = 0
+    all_files = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)]
+    for filename in all_files:
+        with open(filename) as f:
+            content = f.read()
+            total_files_checked += 1
+            for pattern in BAD_PATTERNS:
+                if pattern in content:
+                    summary_messages.append(_MESSAGE_TYPE_FAILED)
```

Comment:
If two or more files fail pattern checks, don't you want to just add a single summary message at the end?

Also, add a suffix to the summary message, similar to the other two cases ("Python linting failed", etc.), so that we can see where the overall failure is from.